[2023-01-21T12:20:43.208+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: real_estate_data_pipeline.clean_and_transform_data manual__2023-01-21T12:20:39.899962+00:00 [queued]>
[2023-01-21T12:20:43.231+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: real_estate_data_pipeline.clean_and_transform_data manual__2023-01-21T12:20:39.899962+00:00 [queued]>
[2023-01-21T12:20:43.232+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2023-01-21T12:20:43.233+0000] {taskinstance.py:1363} INFO - Starting attempt 1 of 2
[2023-01-21T12:20:43.234+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2023-01-21T12:20:43.254+0000] {taskinstance.py:1383} INFO - Executing <Task(PythonOperator): clean_and_transform_data> on 2023-01-21 12:20:39.899962+00:00
[2023-01-21T12:20:43.266+0000] {standard_task_runner.py:54} INFO - Started process 1888 to run task
[2023-01-21T12:20:43.308+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'real_estate_data_pipeline', 'clean_and_transform_data', 'manual__2023-01-21T12:20:39.899962+00:00', '--job-id', '1209', '--raw', '--subdir', 'DAGS_FOLDER/real_estate_dag.py', '--cfg-path', '/tmp/tmpziqii895']
[2023-01-21T12:20:43.311+0000] {standard_task_runner.py:83} INFO - Job 1209: Subtask clean_and_transform_data
[2023-01-21T12:20:43.314+0000] {dagbag.py:525} INFO - Filling up the DagBag from /opt/***/dags/real_estate_dag.py
[2023-01-21T12:20:43.906+0000] {task_command.py:384} INFO - Running <TaskInstance: real_estate_data_pipeline.clean_and_transform_data manual__2023-01-21T12:20:39.899962+00:00 [running]> on host 9667f615f42a
[2023-01-21T12:20:43.974+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=me
AIRFLOW_CTX_DAG_ID=real_estate_data_pipeline
AIRFLOW_CTX_TASK_ID=clean_and_transform_data
AIRFLOW_CTX_EXECUTION_DATE=2023-01-21T12:20:39.899962+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-01-21T12:20:39.899962+00:00
[2023-01-21T12:20:43.986+0000] {logging_mixin.py:117} INFO - The value of the dataframe 1 is :
[2023-01-21T12:20:43.987+0000] {logging_mixin.py:117} INFO - {'address': {'0': '1234 Elm St', '1': '5678 Oak Ave', '2': '1234 Birch Rd'}, 'city': {'0': 'Anytown', '1': 'Othertown', '2': 'Sometown'}, 'zipcode': {'0': 12345, '1': 67890, '2': 54321}, 'price': {'0': 500000, '1': 450000, '2': 425000}, 'bedrooms': {'0': 3, '1': 4, '2': 3}, 'bathrooms': {'0': 2.0, '1': 2.5, '2': 2.0}, 'sqft': {'0': 2000, '1': 2500, '2': 1500}, 'sale_date': {'0': '2022-01-01', '1': '2022-01-02', '2': '2022-01-03'}}
[2023-01-21T12:20:43.988+0000] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 193, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/real_estate_dag.py", line 51, in clean_and_transform_data
    df =  pd.DataFrame.from_dict(json.loads(df), orient='index')
  File "/usr/local/lib/python3.7/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not dict
[2023-01-21T12:20:44.000+0000] {taskinstance.py:1406} INFO - Marking task as UP_FOR_RETRY. dag_id=real_estate_data_pipeline, task_id=clean_and_transform_data, execution_date=20230121T122039, start_date=20230121T122043, end_date=20230121T122044
[2023-01-21T12:20:44.012+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 1209 for task clean_and_transform_data (the JSON object must be str, bytes or bytearray, not dict; 1888)
[2023-01-21T12:20:44.063+0000] {local_task_job.py:164} INFO - Task exited with return code 1
[2023-01-21T12:20:44.105+0000] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
